src/lib/matrix4.js

```diff
@@ -272,6 +272,30 @@ class Matrix4 extends Float32Array {
     return this;
   }
 
+  /**
+   * Scale this matrix by the given factor.
+   *
+   * This implementation is compatible with C44Matrix::Scale(float).
+   *
+   * @param {Number} factor Scaling factor
+   * @returns {Matrix4} Self
+   */
+  scaleByNumber(factor) {
+    this[0]  *= factor;
+    this[1]  *= factor;
+    this[2]  *= factor;
+
+    this[4]  *= factor;
+    this[5]  *= factor;
+    this[6]  *= factor;
+
+    this[8]  *= factor;
+    this[9]  *= factor;
+    this[10] *= factor;
```

Comment:
I understand this 😭 👏

Reply:
Haha, yeah, just rescaling the upper left 3x3 portion of the matrix. :)